packages/shorebird_cli/lib/src/commands/release_new/releaser.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:shorebird_cli/src/commands/release_new/release_type.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+import 'package:shorebird_code_push_protocol/shorebird_code_push_protocol.dart';
+
+/// {@template releaser}
+/// Executes platform-specific functionality to create a release.
+/// {@endtemplate}
+abstract class Releaser {
+  /// {@macro releaser}
+  Releaser({
+    required this.argResults,
+    required this.flavor,
+    required this.target,
+  });
+
+  /// The arguments passed to the command.
+  final ArgResults argResults;
+
+  /// The flavor of the release, if any.
+  final String? flavor;
+
+  /// The target script to run, if any.
+  final String? target;
+
+  /// Whether --release-version must be specified to patch. Currently only
+  /// required for add-to-app/hybrid releases (aar and ios-framework).
+  bool get requiresReleaseVersionArg => false;
```

Comment:
If this is only for patches why is it on the releaser?

Reply:
Bad comment – aar and ios-framework releases require a release version. Addressed!